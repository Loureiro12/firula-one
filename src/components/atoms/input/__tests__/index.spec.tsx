import { render, fireEvent, act } from '@testing-library/react-native';
import { Input } from '../';
import React from 'react';

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "Ionicons",
}));

describe('Input Component', () => {
  const baseProps = {
    label: 'Test Label',
    placeholder: 'Enter text',
    onChangeText: jest.fn(),
    testIDPrefix: 'login',
  };

  it('renders correctly with basic props', () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <Input {...baseProps} />
    );
    
    expect(getByText('Test Label')).toBeTruthy();
    expect(getByPlaceholderText('Enter text')).toBeTruthy();
    expect(getByTestId('login-input-container')).toBeTruthy();
  });

  it('shows error message when provided', () => {
    const { getByText } = render(
      <Input {...baseProps} errorMensage="Invalid input" />
    );
    
    expect(getByText('Invalid input')).toBeTruthy();
  });

  it('renders left icon when provided', () => {
    const { queryByTestId } = render(
      <Input {...baseProps} iconLeft="search" />
    );
    
    expect(queryByTestId('login-input-container')).toHaveProp('children');
  });

  it('renders right icon when provided', () => {
    const { queryByTestId } = render(
      <Input {...baseProps} iconRight="close" />
    );
    
    expect(queryByTestId('login-input-container')).toHaveProp('children');
  });

  describe('Password Input', () => {
    it('toggles password visibility', async () => {
      const { getByTestId } = render(
        <Input {...baseProps} isPassword />
      );
      
      const input = getByTestId('text-input');
      expect(input.props.secureTextEntry).toBe(true);
      
      await act(async () => {
        fireEvent.press(getByTestId('login-toggle-password'));
      });
      
      expect(input.props.secureTextEntry).toBe(false);
    });

    it('disables toggle when input is disabled', async () => {
      const { getByTestId } = render(
        <Input {...baseProps} isPassword isDisabled />
      );
      
      const toggle = getByTestId('login-toggle-password');
      expect(toggle.props.accessibilityState.disabled).toBe(true);
    });
  });

  it('is disabled when isDisabled is true', () => {
    const { getByTestId } = render(
      <Input {...baseProps} isDisabled />
    );
    
    const input = getByTestId('text-input');
    expect(input.props.editable).toBe(false);
  });

  it('calls onChangeText when text is entered', () => {
    const { getByTestId } = render(
      <Input {...baseProps} />
    );
    
    const input = getByTestId('text-input');
    fireEvent.changeText(input, 'new text');
    
    expect(baseProps.onChangeText).toHaveBeenCalledWith('new text');
  });

  it('renders correctly without testIDPrefix', () => {
    const { queryByTestId } = render(
      <Input {...baseProps} testIDPrefix={undefined} />
    );
    
    expect(queryByTestId('input-container')).toBeNull();
  });

  it('matches snapshot with basic props', () => {
    const tree = render(
      <Input {...baseProps} />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});