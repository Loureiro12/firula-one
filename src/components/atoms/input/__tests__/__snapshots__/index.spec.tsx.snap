// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches snapshot with all props 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
  testID="login-input-container"
>
  <Text
    style={
      {
        "color": "#171717",
        "fontSize": 14,
        "fontWeight": "400",
        "marginBottom": 10,
      }
    }
  >
    Test Label
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "gray",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "gray",
            "fontSize": 24,
          },
          {
            "marginRight": 8,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <TextInput
      editable={true}
      onChangeText={
        [MockFunction] {
          "calls": [
            [
              "new text",
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="Enter text"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "height": "100%",
        }
      }
      testID="text-input"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "gray",
              "fontSize": 24,
            },
            {
              "marginLeft": 8,
            },
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "gray",
            "fontSize": 24,
          },
          {
            "marginLeft": 8,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#ef4444",
        "fontSize": 12,
        "marginTop": 4,
      }
    }
  >
    Error message
  </Text>
</View>
`;

exports[`Input Component matches snapshot with basic props 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
  testID="login-input-container"
>
  <Text
    style={
      {
        "color": "#171717",
        "fontSize": 14,
        "fontWeight": "400",
        "marginBottom": 10,
      }
    }
  >
    Test Label
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "gray",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 12,
      }
    }
  >
    <TextInput
      editable={true}
      onChangeText={
        [MockFunction] {
          "calls": [
            [
              "new text",
            ],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      placeholder="Enter text"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "height": "100%",
        }
      }
      testID="text-input"
    />
  </View>
</View>
`;
